test/builtin_dataset_mocks.py

```diff
@@ -470,7 +471,10 @@ def imagenet(info, root, config):
         ]
         num_children = 1
         synsets.extend((0, "", "", "", num_children, [], 0, 0) for _ in range(5))
-        savemat(data_root / "meta.mat", dict(synsets=synsets))
+        with warnings.catch_warnings():
```

Comment:
I'm not sure we should do that, if there is ever an actual deprecation warning for savemat we would miss it

Reply:
> Thus, unless scipy.io.savemat emits a np.VisibleDeprecationWarning it will just be raised as it was before.

Yes, this is what I was concerned about. scipy does raise these warnings on their own utils sometimes (perhaps only on some very specific cases, IDK).  https://github.com/scipy/scipy/search?q=np.VisibleDeprecationWarning

Reply:
But I did not realize that this was a test, sorry, I thought this was dataset code.

What tripped me up is that we usually use `@pytest.mark.filterwarnings` to ignore specific warnings in some specific tests, instead of using the `catch_warning` context manager. I assume that because of our custom dataset testing framework, that kind of decorator-based filtering would be tricky to do on a per-dataset basis?